src/main/java/org/aerogear/kryptowire/BinaryHistoryAction.java

```diff
@@ -78,6 +80,13 @@ public BinaryStatus getStatus() throws IOException, InterruptedException {
             return BinaryStatus.notReady();
         }
         this.status = BinaryStatus.fromJSONObject(out);
+
+        File targetFile = new File(getRun().getArtifactsDir() + "/kryptowire.pdf");
+        kws.downloadReport(this.info.getHash(), "pdf", targetFile);
+
+        targetFile = new File(getRun().getArtifactsDir() + "/kryptowire-niap.pdf");
```

Comment:
I think we should check if the target file exists and only download the report if it doesn't